sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -436,7 +435,8 @@ private async Task RenewOwnershipAsync(CancellationToken cancellationToken)
 
             Logger.RenewOwnershipStart(OwnerIdentifier);
 
-            IEnumerable<EventProcessorPartitionOwnership> ownershipToRenew = InstanceOwnership.Values
+            List<EventProcessorPartitionOwnership> ownershipToRenew = InstanceOwnership.Values
```

Comment:
Can you walk me through your intent here?   The way that I read it is "renew anything that has a last modified date further in the future than the load balancing interval."    I'm not sure if I'm just not following or the calculation may be reversed.

For example:

```
Now =  2pm, 30 seconds
Last Modified = 2pm, 0 seconds

(Last Modified - Now) =  -30 seconds
```

Reversed:
```
Now =  2pm, 30 seconds
Last Modified = 2pm, 0 seconds

(Now - Last Modified) =  30 seconds
```

I don't see how we'd have a last modified date that was greater than `Now`,  so we would never be greater than the load balancing interval, would we?

I'm also not sure how that is intended to play with the expiration time.   If we have a last modified stamp that will expire before the next load balancing cycle, we definitely want to renew.... and I think that would have to use the OwnershipExpirationInterval to calculate rather than just the straight use of `now?     

Without prototyping deeply, I think this would come out something like:

```csharp
// Simulate 30 second expiration, 10 second load balancing interval.
var expiration = TimeSpan.FromSeconds(30);
var cycle = TimeSpan.FromSeconds(10);

// Simulate a last modified stamp of 10 seconds ago.
var now = DateTimeOffset.UtcNow;
var last = now.Subtract(TimeSpan.FromSeconds(10));
	
// This assumes that we want to flag something that expires before the next load balancing 
// cycle.  We may want to make this closer to "this thing expires within the next X cycles.
var expires = last.Add(expiration);
var nextCycle = now.Add(cycle);
var shouldRenew = (expires <= nextCycle);
```

To allow for fault tolerance, like we do today, we would probably want to extend that a bit to say "If this expires within the next X number of load balancing intervals then update it.

Reply:
Plop. You are right, the condition is wrong and I need a better test.